kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ParquetStatsReader.java

```diff
@@ -0,0 +1,238 @@
+
+        Map<Column, Literal> minValues = new HashMap<>();
+        Map<Column, Literal> maxValues = new HashMap<>();
+        Map<Column, Long> nullCounts = new HashMap<>();
+        for (Column statsColumn : statsColumns) {
+            Optional<Statistics<?>> stats = statsForColumn.get(statsColumn);
+            DataType columnType = getDataType(dataSchema, statsColumn);
+            if (!stats.isPresent() || !isStatsSupportedDataType(columnType)) {
+                continue;
+            }
+            Statistics<?> statistics = stats.get();
+
+            Literal minValue = decodeMinMaxStat(columnType, statistics, true /* decodeMin */);
+            minValues.put(statsColumn, minValue);
+
+            Literal maxValue = decodeMinMaxStat(columnType, statistics, false /* decodeMin */);
+            maxValues.put(statsColumn, maxValue);
+
+            Long numNulls = statistics.isNumNullsSet() ? statistics.getNumNulls() : null;
+            nullCounts.put(statsColumn, numNulls);
+        }
+
+        return new DataFileStatistics(rowCount, minValues, maxValues, nullCounts);
+    }
+
+    private static Literal decodeMinMaxStat(
+            DataType dataType,
+            Statistics<?> statistics,
+            boolean decodeMin) {
+        Object statValue = decodeMin ? statistics.genericGetMin() : statistics.genericGetMax();
+        if (statValue == null) {
+            return null;
+        }
+
+        if (dataType instanceof ByteType) {
+            return Literal.ofByte(((Number) statValue).byteValue());
+        } else if (dataType instanceof ShortType) {
+            return Literal.ofShort(((Number) statValue).shortValue());
+        } else if (dataType instanceof IntegerType) {
+            return Literal.ofInt(((Number) statValue).intValue());
+        } else if (dataType instanceof LongType) {
+            return Literal.ofLong(((Number) statValue).longValue());
+        } else if (dataType instanceof FloatType) {
+            return Literal.ofFloat(((Number) statValue).floatValue());
+        } else if (dataType instanceof DoubleType) {
+            return Literal.ofDouble(((Number) statValue).doubleValue());
+        } else if (dataType instanceof DecimalType) {
+            LogicalTypeAnnotation logicalType = statistics.type().getLogicalTypeAnnotation();
+            checkArgument(
+                    logicalType instanceof DecimalLogicalTypeAnnotation,
+                    "DECIMAL column had invalid Parquet Logical Type: %s", logicalType);
+            int scale = ((DecimalLogicalTypeAnnotation) logicalType).getScale();
+
+            DecimalType decimalType = (DecimalType) dataType;
+
+            // Check the scale is same in both the Delta data type and the Parquet Logical Type
+            checkArgument(
+                    scale == decimalType.getScale(),
+                    "DECIMAL column has different scale compared to the input data: %s", scale);
```

Comment:
`DECIMAL column` and `input` seem a little abstract to me ..... can we say "physical" vs "logical" or be a bit more clear?